tests/test_matching_algorithms/test_requirements_validators.py

```diff
@@ -0,0 +1,120 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+from pendulum import now
+
+from d3a_interface.data_classes import Offer, Bid
+from d3a_interface.matching_algorithms.requirements_validators import (
+    EnergyTypeRequirement,
+    TradingPartnersRequirement,
+    RequirementsSatisfiedChecker, SelectedEnergyRequirement, PriceRequirement)
+
+
+@pytest.fixture
+def offer():
+    return Offer("id", now(), 2, 2, "other", 2)
+
+
+@pytest.fixture
+def bid():
+    return Bid("bid_id", now(), 9, 10, "B", 9)
+
+
+class TestRequirementsValidator:
+    """Test all bid/offer requirements validators."""
+
+    def test_trading_partners_requirement(self, offer, bid):
```

Comment:
It would be good to split all these tests into separate test cases. At the moment, you're grouping tests for failures with tests for success. Ideally, each scenario should be described by just one test.
I would suggest to create a class for each individual requirement class (e.g. `TestTradingPartnersRequirement`) and separate individual methods (e.g. `test_is_satisfied_succeeds`, `test_is_satisfied_<condition>_fails`).

Even without going so much into the granular level, it would still be better to split these tests a bit.

Reply:
Since unit tests focus on a single function at a time, I usually recommend following a specific structure for naming and testing. Each method/function should be tested, so the name of the test could be something like: `test_<function-name>_<condition>_<expected-output>`. Following this makes it easier to create one (or more) tests for each individual function.